File: app/src/main/java/rssdemo/andras/hu/rssdemo/ui/subscriptions/SubscriptionsViewModel.java
package rssdemo.andras.hu.rssdemo.ui.subscriptions;


import rssdemo.andras.hu.rssdemo.repository.SubscriptionRepository;

public class SubscriptionsViewModel {

    private SubscriptionRepository subscriptionRepository;
    private SubscriptionsView view;

    public SubscriptionsViewModel(SubscriptionRepository subscriptionRepository) {
        this.subscriptionRepository = subscriptionRepository;
    }

    public void setView(SubscriptionsView view) {
        this.view = view;
        refreshData();
    }

    void refreshData() {
        view.getAdapter().setItems(subscriptionRepository.getSubscriptions());
    }

    void deleteSubscription(String name) {
        subscriptionRepository.delete(name);
        refreshData();
    }
}

File: app/src/main/java/rssdemo/andras/hu/rssdemo/ui/subscriptions/SubscriptionsViewModel.kt
package rssdemo.andras.hu.rssdemo.ui.subscriptions


import rssdemo.andras.hu.rssdemo.repository.SubscriptionRepository

class SubscriptionsViewModel(private val subscriptionRepository: SubscriptionRepository) {

    private lateinit var view: SubscriptionsView

    internal fun setView(view: SubscriptionsView) {
        this.view = view
        refreshData()
    }

    internal fun refreshData() {
        view.setSubscriptionItems(subscriptionRepository.subscriptions)
    }

    internal fun deleteSubscription(name: String) {
        subscriptionRepository.delete(name)
        refreshData()
    }
}
